logger_block.py

```diff
@@ -33,11 +31,21 @@ def process_signals(self, signals):
             None
         """
         log_func = self._get_logger()
-        for s in signals:
+
+        # make it obvious that this was passed a list of signals or just one
+        # by logging a list or not.
+        if len(signals) > 1:
             try:
-                log_func(s)
+                log_func(signals)
             except:
-                self.logger.error("Failed to log signal")
+                self.logger.error("Failed to log list of signals: {}"
+                                  .format(signals))
+        else:
+            for signal in signals:
```

Comment:
I don't like that this has a loop when we already know that signals is not greater than 1.